test/lifecycles/attributes.test.js

```diff
@@ -0,0 +1,69 @@
+import define from "../../lib/define";
+import { onAttributeChanged } from "../../lib/lifecycles/index.js";
+
+describe("onAttributeChange", function() {
+	beforeEach(() => {
+		const
+			Component = () => () => `<span>dummy text content</span>`;
+		
+		Component.observedAttributes = ['foo'];
+		define("attribute-test-element", Component);
+		
+		const
+			element = document.createElement("attribute-test-element");
+		element.setAttribute("foo", "init");
+      document.body.appendChild(element);
+	});
+	
+	afterEach(() => {
+		document.querySelector('attribute-test-element').remove();
+	});
+	
+	it("should call the callback with the assigned attribute value", (done) => {
+		const
+			element = document.querySelector('attribute-test-element');
+      
+      onAttributeChanged(element, ({name, current, old}) => {
+      	if (name !== "foo") {
```

Comment:
We could use a mock function and check if that function was called or not called in the other test. In that way the test will be simpler and we don't need to throw errors

Reply:
Sorry, I don't get it 🤔. Could you describe it more?